### PR TITLE
Feature/meme creator

### PR DESCRIPTION
**Pull request by ClaytonVis**
This should have the previous merge revert sorted out!

This branch got carried away and is actually the majority of the meme creation center. There is still a bit I need to do:

- EDIT: DONE - (most important) I haven't fully implemented the functionality for the 'Publish Meme'
    - I have set-up a service to interact with the Imgur api, it still needs to be connected to Meme Service.
    - I also need to make a search bar to select which community the meme is going to be published to. (@Step7750 I'm hoping to be able to re-use your search bar, having to filter out the users in the results)
- Color picker for TextBox (for shadow/stroke as well).
    - I looked at some packages on npm, but was having some dependency problems, so I moved on to more pressing issues.
- The interface to edit an already placed fabric object (i.e. changing a TextBox's font/size/etc.)
- The interface to view all objects already in the canvas (ordered by depth) allowing selection by clicking on the list label.

The following two will not make the deadline, but are major issues nonetheless:
- CORS hell... For the time being the URL upload button has been disabled, as a image loaded to the canvas from a different URL taints the canvas and will no longer be able to convert to JSON/blob/data. 
    - While this is a pretty important feature, it's a big task so it's not feasible for the deadline.
- Bottom bar to view templates, which expands to page

Another issue which I'd like some input on, is where we are placing the button to go to the meme creation page (also what to label the button). I currently have it in the SideNav labelled 'Creation Center', but I'm sure someone has better design instincts than myself (@xiningchen I feel like you're the best at this kinda stuff)

P.s. @Step7750 I don't know why Travis CI won't build successfully, I don't know how else to import fabric. This is probably something that needs to be addressed though.

### CODE REVIEW

File: src/app/creation/fabric/fabric.component.ts [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+export class FabricComponent {
+  @ViewChild('canvCont') canvCont;
+  private parent: CreationComponent;
+
+  private canvas;
+  private functComp: FunctionBarComponent;
+  public height: number; width: number;
+  private zoomVal: number; scaledHeight: number; scaledWidth: number;
+
+  private badURL = 'https://fthmb.tqn.com/i5PTJXJQGb5Kg64f2T1mMDt2ULg=/768x0/filters:no_upscale():max_bytes(150000):strip_icc()/ggg-580734603df78cbc28f46d37.PNG';
```
Comment (Step7750) [inline src/app/creation/fabric/fabric.component.ts]: This can be removed

File: src/app/creation/fabric/fabric.component.ts [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+})
+
+export class FabricComponent {
+  @ViewChild('canvCont') canvCont;
+  private parent: CreationComponent;
+
+  private canvas;
+  private functComp: FunctionBarComponent;
+  public height: number; width: number;
+  private zoomVal: number; scaledHeight: number; scaledWidth: number;
```
Comment (Step7750) [inline src/app/creation/fabric/fabric.component.ts]: Put these on separate lines

File: src/app/creation/imgur.service.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+export class ImgurService {
+  private url = 'https://api.imgur.com/3/';
+  private clientID = '2e63301babbf33a';
+
+
+
+  constructor(private http: HttpClient) { }
+
+  uploadImg(dataURL: string): Promise<ImgurUploadResponse> {
+    // TODO: Get the community that the meme is going into
```
Comment (Step7750) [inline src/app/creation/imgur.service.ts]: This can be removed

File: src/app/sidenav/sidenav.component.html [hunk 4]
```diff
@@ -4,6 +4,8 @@
     <mat-nav-list>
       <a mat-list-item [routerLink]="['/create-community']">Create Community</a>
       <mat-divider></mat-divider>
+      <a mat-list-item [routerLink]="['/creation']">Create Memes</a>
```
Comment (Step7750) [inline src/app/sidenav/sidenav.component.html]: We should only display this if they're logged in, `*ngIf="userService.isLoggedIn()"`

Comment (ClaytonVis) [reply src/app/sidenav/sidenav.component.html]: I disagree with that, someone who has never signed up might not even know about the meme creation functionality if the button isn't accessible. If anything they should be allowed to create a meme, and be prompted to log in/sign up as soon as they try to publish/download

File: package.json [hunk 5]
```diff
@@ -25,13 +25,16 @@
     "@angular/router": "^5.2.0",
     "core-js": "^2.4.1",
     "rxjs": "^5.5.6",
+    "fabric": "^2.1.0",
+    "hammerjs": "^2.0.8"
```
Comment (Step7750) [inline package.json]: Missing comma, causing failed build

**Pull request merged**